rllib/metrics/metric.py

```diff
@@ -0,0 +1,88 @@
+from abc import abstractmethod
+from typing import Dict
+
+
+class Metric:
```

Comment:
IIUC this is more of a CallbacksWrapper (similar to the Keras callbacks class). Since the callbacks can be used for more than recording metrics (e.g., can be used for curriculum learning and other ad hoc model updates), how about we do the following renaming:
- Metric => Callback
- MetricsWrapper => CallbacksList

I think this would make it more clear for users. We can even point to https://keras.io/callbacks/ for conceptual guidance.